### DISCUSSION TITLE
General approach to upgrade from HEAD / beta versions

### DISCUSSION THREAD

**Discussion by johnalanwoods** [Homebrew/discussions#414] [Everyday usage]
This is not specifically related to Homebrew on Apple Silicon, but in general...

When I use Homebrew to install with `--HEAD` or indeed, if Homebrew installs from source applying some random patch (as it has for beta support for some formula on Apple ARM64, does this preclude normal/regular updates in future?

i.e. if I build a formula from src/HEAD, or I have installed an early support version for ARM64, and some months later I run `brew upgrade/update`, will brew just update me to the latest version without issue?

Or, a bit like macOS beta, once your on the beta track are you precluded from regular updates?

**Comment by SMillerDev** [comment]
As far as I know, HEAD builds don't update. You'll have to reinstall a head build to get it to build the state of the art at the time.

**Reply by carlocab** [reply]
Yep. `brew upgrade` ignores HEAD builds. To check whether a HEAD build is outdated, you can do

    brew outdated --fetch-HEAD $formula

and `brew reinstall` to upgrade it.

**Reply by infa-jowoods** [reply]
Thanks!! What about “beta”/early support formula?

**Reply by carlocab** [reply]
What is a

>beta”/early support formula

?

**Reply by johnalanwoods** [reply]
like Rust 1.49, or JDK 15 with Apple Silicon patch... that kinda thing, patched formula?

Also is there a way to list any HEAD formula that was installed? Not Outdated, but just generally installed?

Thanks @carlocab

**Reply by carlocab** [reply]
You don't need to use `--HEAD` to install those, so you can treat them exactly as if they were a normal formula.

As for `HEAD` formulae installed, I would do

    brew list --versions | grep HEAD

**Reply by SMillerDev** [reply]
> like Rust 1.49, or JDK 15 with Apple Silicon patch... that kinda thing, patched formula?

Rust 1.49 is GA and we didn't patch the JDK to work with Apple silicon yet. But when we do it's just a normal formula like any other that just happens to work on ARM.

**Reply by infa-jowoods** [reply]
Thank you all for your time and knowledge. Greatly appreciated.
